fiftyone/plugins/utils.py

```diff
@@ -219,3 +222,103 @@ def _do_get_plugin_info(task):
             spec = gh_repo
 
         logger.debug("Failed to retrieve plugin info for %s: %s", spec, e)
+
+
+def get_target_view(ctx, target):
+    """
+    Get the target view for the current context. This is a helper function that
+    returns the view based on the `target` parameter. To be used in conjunction
+    with `list_target_views()`.
+
+    Args:
+        ctx: the current context
+        target: the target view
+
+    Returns:
+        the target view
+
+    """
+    if target == "SELECTED_SAMPLES":
+        return ctx.view.select(ctx.selected)
+
+    if target == "DATASET":
+        return ctx.dataset
+
+    return ctx.view
+
+
+def list_target_views(ctx, inputs):
+    """
+    List the available target views for the current context. This is a helper
+    function that sets the `target` parameter based on the current context. Use
+    it in the `inputs` method of a plugin to allow the user to select the target
+    view from selected samples, the current view, or the entire dataset. To be
+    used in conjunction with `get_target_view()`.
+
+    Args:
+        ctx: the current context
+        inputs: the inputs object
+
+    Returns:
+        None
+
+    """
+    has_view = ctx.view != ctx.dataset.view()
+    has_selected = bool(ctx.selected)
+    default_target = "DATASET"
+    if has_view or has_selected:
+        target_choices = types.RadioGroup()
+        target_choices.add_choice(
+            "DATASET",
+            label="Entire dataset",
+            description="Run on the entire dataset",
+        )
+
+        if has_view:
+            target_choices.add_choice(
+                "CURRENT_VIEW",
+                label="Current view",
+                description="Run on the current view",
+            )
+            default_target = "CURRENT_VIEW"
+
+        if has_selected:
+            target_choices.add_choice(
+                "SELECTED_SAMPLES",
+                label="Selected samples",
+                description="Run on the selected samples",
+            )
+            default_target = "SELECTED_SAMPLES"
+
+        inputs.enum(
+            "target",
+            target_choices.values(),
+            default=default_target,
+            view=target_choices,
+        )
+    else:
+        ctx.params["target"] = "DATASET"
+
+
+def _serialize_view(view):
+    return json.loads(json_util.dumps(view._serialize()))
+
+
+def set_view(ctx, view):
```

Comment:
I had a plan for these built in operators. The usage I want to include is:

```
ctx.ops.set_view(view)
```

In general I would like to organize things around the `ExecutionContext` and `operator.types`.

Reply:
Alternatively, what if we just supported this?

```py
ctx.trigger("set_view", view=view)
```

with the idea that the builtin serialization logic inside the core operator framework would check for `DatasetView` objects and automatically serialize them via `json.loads(json_util.dumps(view._serialize()))`?

Reply:
That works for me, but is it okay to just accept a non-serialized object in one specific case?

Reply:
I don't want `trigger()` to be more magical. `ctx.ops` is a great place for these things (eg preparing to a `ctx.trigger()` call). It also removes the "set_view" magic string.